packages/legend-extension-dsl-service/src/components/studio/ServiceRegisterModal.tsx

```diff
@@ -0,0 +1,442 @@
+                      onInputChange={onSearchTextChange}
+                      onChange={onUserOptionChange}
+                      isMulti={true}
+                      allowCreating={true}
+                      value={selectedOwners}
+                    />
+                  </div>
+                </div>
+                <div className="service-register-modal__input">
+                  <div className="service-register-modal__input__label">
+                    Execution Server
+                  </div>
+                  <div className="input-group service-register-modal__input__selector">
+                    <CustomSelectorInput
+                      options={envOptions}
+                      onChange={onServerEnvChange}
+                      value={selectedEnvOption}
+                      darkMode={true}
+                    />
+                  </div>
+                </div>
+                <div
+                  className="service-register-modal__auto-activation__toggler"
+                  onClick={toggleActivateService}
+                >
+                  <div className="panel__content__form__section__toggler">
+                    <button
+                      className={clsx(
+                        'panel__content__form__section__toggler__btn',
+                        {
+                          'panel__content__form__section__toggler__btn--toggled':
+                            activateService,
+                        },
+                      )}
+                      tabIndex={-1}
+                    >
+                      {activateService ? <CheckSquareIcon /> : <SquareIcon />}
+                    </button>
+                    <div className="panel__content__form__section__toggler__prompt">
+                      Activate service after registration
+                    </div>
+                  </div>
+                </div>
+              </div>
+            </PanelFullContent>
+          </Panel>
+          <div className="search-modal__actions">
+            <button className="btn btn--dark" onClick={registerService}>
+              Register Service
+            </button>
+            <button className="btn btn--dark" onClick={onClose}>
+              Close
+            </button>
+          </div>
+        </Modal>
+      </Dialog>
+    );
+  },
+);
+
+export const ServiceRegisterAction = observer(
+  (props: { editorStore: ExistingQueryEditorStore }) => {
+    const { editorStore } = props;
+    const [showRegisterServiceModal, setShowRegisterServiceModal] =
+      useState(false);
+    const registerCurrentQuery = (): void => {
+      setShowRegisterServiceModal(true);
+    };
+    const onClose = (): void => setShowRegisterServiceModal(false);
+    return (
+      <>
+        <button
+          className="query-editor__header__action btn--dark"
+          tabIndex={-1}
+          onClick={registerCurrentQuery}
+          title="Register query as service..."
+        >
+          <RobotIcon />
```

Comment:
The main action here is register so let's change the icon to the rocket icon which symbolizes the register.